apps/studio/components/grid/components/header/RefreshButton.tsx

```diff
@@ -6,14 +6,14 @@ import { tableRowKeys } from 'data/table-rows/keys'
 import { Button } from 'ui'
 
 export type RefreshButtonProps = {
-  table: { id: number }
+  tableId?: number
   isRefetching?: boolean
 }
 
-const RefreshButton = ({ table, isRefetching }: RefreshButtonProps) => {
+const RefreshButton = ({ tableId, isRefetching }: RefreshButtonProps) => {
   const { ref } = useParams()
   const queryClient = useQueryClient()
-  const queryKey = tableRowKeys.tableRowsAndCount(ref, table.id)
+  const queryKey = tableRowKeys.tableRowsAndCount(ref, tableId)
```

Comment:
likewise here - wondering if we can just use id from `useParams`

Reply:
this one seems to be passed in from a foreign key:

<img width="786" alt="Screenshot 2025-03-21 at 15 16 56" src="https://github.com/user-attachments/assets/68c32cf7-dff6-405e-958e-788f06196be0" />